bundles/org.openhab.binding.openthermgateway/src/main/java/org/openhab/binding/openthermgateway/internal/OpenThermGatewaySocketConnector.java

```diff
@@ -0,0 +1,227 @@
+/**
+ * Copyright (c) 2010-2018 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.openthermgateway.internal;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.PrintWriter;
+import java.net.InetSocketAddress;
+import java.net.Socket;
+import java.util.AbstractMap;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Map.Entry;
+
+import org.eclipse.smarthome.core.types.State;
+import org.openhab.binding.openthermgateway.handler.TypeConverter;
+
+/**
+ * The {@link OpenThermGatewaySocketConnector} is responsible for handling the socket connection
+ *
+ * @author Arjen Korevaar - Initial contribution
+ * @author Arjan Mels - Improved robustness by re-sending commands, handling all message types (not only Boiler)
+ */
+public class OpenThermGatewaySocketConnector implements OpenThermGatewayConnector {
```

Comment:
maybe rename to `OpenThermGatewayConnectorImpl` to make it more clear what this class is doing

Reply:
The 'Socket' part refers to a TCP Socket implementation. Future implementations may be named SerialConnector. Impl is too generic and won't indicate the kind of connector that is implemented.